airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockDestinationDirectLoader.kt

```diff
@@ -22,15 +24,25 @@ import kotlinx.coroutines.runBlocking
 
 @Singleton
 @Requires(env = [MOCK_TEST_MICRONAUT_ENVIRONMENT])
-class MockDestinationDirectLoaderFactory : DirectLoaderFactory<MockDestinationDirectLoader> {
+class MockDestinationDirectLoaderFactory(
+    @Named("dataChannelFormat") private val dataChannelFormat: DataChannelFormat
+) : DirectLoaderFactory<MockDestinationDirectLoader> {
     override fun create(streamDescriptor: DestinationStream.Descriptor, part: Int) =
-        MockDestinationDirectLoader()
+        MockDestinationDirectLoader(dataChannelFormat)
 }
 
-class MockDestinationDirectLoader : DirectLoader {
+class MockDestinationDirectLoader(
+    private val dataChannelFormat: DataChannelFormat,
+) : DirectLoader {
     override suspend fun accept(record: DestinationRecordRaw): DirectLoader.DirectLoadResult {
         val recordAirbyteValue = record.asDestinationRecordAirbyteValue()
         val filename = getFilename(record.stream.mappedDescriptor, staging = true)
+        val dataChannelFormatInducedChanges =
+            if (dataChannelFormat == DataChannelFormat.PROTOBUF) {
+                record.stream.unknownColumnChanges
+            } else {
+                emptyList()
+            }
```

Comment:
this feels a bit sketchy, but :shrug: seems like this is how things are intended to work for now